src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedLocal.cs

```diff
@@ -48,13 +50,16 @@ internal sealed class SynthesizedLocal : LocalSymbol
             Debug.Assert(!type.IsVoidType());
             Debug.Assert(!kind.IsLongLived() || syntaxOpt != null);
             Debug.Assert(refKind != RefKind.Out);
+            Debug.Assert((valEscapeScope == null && kind != SynthesizedLocalKind.InterpolatedStringBuilder) ||
```

Comment:
Hmm, so it looks like we are only allowing escape for string builders. This feels... clunky? to me? That we have a single type that reprsents two different things. Although I'm not sure if it's worth splitting it up or not...